### PR TITLE
[ADD][account_tax_cash_basis_fix_currency_rate] - Use of analytic acc…

### PR DESCRIPTION
**Pull request by ougc27**
…ounts in the exchange difference.

### CODE REVIEW

**Comment by ougc27** [comment]
@alan196 Could you check this Pull Request?.

This PR has a improve adding the analytic account to the expense lines of the currency exchange difference journal entry.

**Comment by alan196** [comment]
Check gitignore file you are adding .pyc files

File: account_tax_cash_basis_fix_currency_rate_difference/models/account_partial_reconcile.py [hunk 1]
```diff
@@ -29,11 +29,37 @@ def create_exchange_rate_entry(self, aml_to_fix, amount_diff,
...
                 rec.credit_move_id.journal_id.type == 'purchase' else
                 rec.credit_move_id)
+            lines = []
+            analytic_accounts = []
+
+            # We get the analytic account if this exists in the invoice
+            for line in invoice_move.move_id.line_ids.filtered(
+                    lambda r: r.analytic_account_id):
+                analytic_accounts.append(line.analytic_account_id.id)
+            total_analytic_account = len(analytic_accounts)
```
Comment (alan196) [inline account_tax_cash_basis_fix_currency_rate_difference/models/account_partial_reconcile.py]: you can use `if analytic_accounts:` 

You don't need to use len.

if you test an empty list you will get False.

File: account_tax_cash_basis_fix_currency_rate_difference/models/account_partial_reconcile.py [hunk 2]
```diff
@@ -29,11 +29,37 @@ def create_exchange_rate_entry(self, aml_to_fix, amount_diff,
...
+            if total_analytic_account > 0:
+                analytic_accounts = list(set(analytic_accounts))[0]
+            else:
+                analytic_accounts = False
```
Comment (alan196) [inline account_tax_cash_basis_fix_currency_rate_difference/models/account_partial_reconcile.py]: why you need a False? If you return an empty list what happens?

File: account_tax_cash_basis_fix_currency_rate_difference/models/account_partial_reconcile.py [hunk 3]
```diff
@@ -29,11 +29,37 @@ def create_exchange_rate_entry(self, aml_to_fix, amount_diff,
...
+            lines = []
+            analytic_accounts = []
+
+            # We get the analytic account if this exists in the invoice
+            for line in invoice_move.move_id.line_ids.filtered(
+                    lambda r: r.analytic_account_id):
+                analytic_accounts.append(line.analytic_account_id.id)
+            total_analytic_account = len(analytic_accounts)
+            if total_analytic_account > 0:
+                analytic_accounts = list(set(analytic_accounts))[0]
```
Comment (alan196) [inline account_tax_cash_basis_fix_currency_rate_difference/models/account_partial_reconcile.py]: What will happen if you get more than 1 analytic account? You will miss it.

File: account_tax_cash_basis_fix_currency_rate_difference/models/account_partial_reconcile.py [hunk 4]
```diff
@@ -29,11 +29,37 @@ def create_exchange_rate_entry(self, aml_to_fix, amount_diff,
...
+            # We get only the Profit / Loss line to put the analytic account
+            for tax in taxes:
+                tax_accounts.append(tax.cash_basis_account.id)
+                tax_accounts.append(tax.account_id.id)
+                tax_accounts.append(tax.refund_account_id.id)
+            for aml in move.line_ids:
+                if (aml.account_id.id not in tax_accounts and
+                        not aml.account_id.reconcile):
+                    lines.append((1, aml.id, {
+                        'analytic_account_id': analytic_accounts,
```
Comment (alan196) [inline account_tax_cash_basis_fix_currency_rate_difference/models/account_partial_reconcile.py]: try to send the empty list, if it fails you can use:

`analytic_accounts if analytic_accounts else False`

**Review by alan196 — CHANGES_REQUESTED** [review]

File: account_tax_cash_basis_fix_currency_rate_difference/models/account_partial_reconcile.py (hunk 3 above)
Comment (ougc27) [reply account_tax_cash_basis_fix_currency_rate_difference/models/account_partial_reconcile.py]: In this case, what is the better way to adapt this field to the journal entry?

File: account_tax_cash_basis_fix_currency_rate_difference/models/account_move.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+# Copyright 2017, Jarsa Sistemas, S.A. de C.V.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from openerp import _, api, fields, models
+
+
+class AccountMove(models.Model):
+    _inherit = 'account.move'
+
+    nbr_diff_moves = fields.Float(
```
Comment (alan196) [inline account_tax_cash_basis_fix_currency_rate_difference/models/account_move.py:11]: Why nbr? Please add help to explain this field.

File: account_tax_cash_basis_fix_currency_rate_difference/models/account_move.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        string='Exchange Diferential', compute="_compute_nbr_diff_moves",
+        default=0,)
+    account_diff_move_ids = fields.One2many(
+        'account.move',
+        'diff_move_id',
+        string='Account Move',
+    )
+    diff_move_id = fields.Many2one(
+        'account.move',
```
Comment (alan196) [inline account_tax_cash_basis_fix_currency_rate_difference/models/account_move.py:20]: Add string to name the field even if the field is not in the view

File: account_tax_cash_basis_fix_currency_rate_difference/models/account_partial_reconcile.py [hunk 7]
```diff
@@ -29,60 +29,120 @@ def create_exchange_rate_entry(self, aml_to_fix, amount_diff,
...
+                            default_debit_account_id.id or
+                            rec.company_id.currency_exchange_journal_id.
+                            default_credit_account_id.id),
+                        'move_id': move.id,
+                        'currency_id': currency.id,
+                        'partner_id': rec.debit_move_id.partner_id.id,
+                        'analytic_account_id': (
+                            an_lines[index]['analytic_account_id'][0]),
+                    }))
+                import ipdb; ipdb.set_trace()
```
Comment (alan196) [inline account_tax_cash_basis_fix_currency_rate_difference/models/account_partial_reconcile.py]: Please delete the ipdb 👎

**Review by alan196 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**